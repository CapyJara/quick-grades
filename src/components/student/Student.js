import React from 'react';
import PropTypes from 'prop-types';
import styles from './student.css';

const Student = ({ id, name, pending, missing, selectStudent, handleStudentSelect }) => {
  return (
    <li className={styles.student} onClick={handleStudentSelect}>
      <button onClick={() => selectStudent({ id, pending, name })}>
        <h2>{name}</h2>
        <span className={styles.missing}>{missing.length}</span>
        <span className={styles.pending} >{pending.length}</span>
      </button>
    </li>
  );
};

Student.propTypes = {
  name: PropTypes.string.isRequired,
  pending: PropTypes.array.isRequired,
  missing: PropTypes.array.isRequired,
  id: PropTypes.number.isRequired,
  selectStudent: PropTypes.func.isRequired,
  handleStudentSelect: PropTypes.func.isRequired
};

export default Student;
