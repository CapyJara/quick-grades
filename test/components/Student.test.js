import React from 'react';
import { shallow } from 'enzyme';
import Student from '../../src/components/student/Student';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Student 
      id={'12345'}
      name={'mike'}
      pending={4}
      missing={2}
      handleStudentSelect={jest.fn()} 
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
