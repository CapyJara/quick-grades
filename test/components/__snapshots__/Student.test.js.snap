// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<li
  className="student"
>
  <button
    onClick={[Function]}
  >
    <h2>
      mike
    </h2>
    <span
      className="missing"
    >
      2
    </span>
    <span
      className="pending"
    >
      4
    </span>
  </button>
</li>
`;
